service/domain/transport/rpc/request_stream.go

```diff
@@ -98,12 +115,35 @@ func (rs *RequestStream) TerminatedByRemote() {
 	rs.cancel()
 }
 
-func (rs *RequestStream) HandleNewMessage(msg *transport.Message) error {
+func (rs *RequestStream) HandleNewMessage(msg transport.Message) error {
 	if rs.typ != ProcedureTypeDuplex {
-		return errors.New("only duplex streams can receive more than one message")
+		return errors.New("only duplex streams can receive messages")
 	}
 
-	// todo pass msg to the handler
+	select {
+	case <-rs.ctx.Done():
+		return rs.ctx.Err()
+	case rs.incomingMessagesIn <- IncomingMessage{
+		Body: msg.Body,
+	}:
+		return nil
+	}
+}
 
-	return nil
+func (rs *RequestStream) passIncomingMessages() {
```

Comment:
This avoids a stupid situation where we want to both close an output channel and also write to the output channel and end up writing to a closed channel. I don't know how else to synchronize this without making a mess with extra fields and an extra mutex.

Reply:
In general I think lets just ignore stuff like this for now and worry about refactoring it later.